include/readdy/kernel/singlecpu/SCPUStateModel.h

```diff
@@ -45,6 +45,10 @@ namespace scpu {
 
 class SCPUStateModel : public readdy::model::KernelStateModel {
     using topology_action_factory = readdy::model::top::TopologyActionFactory;
+    using particle_t = readdy::model::Particle;
+    using reaction_counts_order1_map = std::unordered_map<particle_t::type_type, std::vector<std::size_t>>;
```

Comment:
You have defined this typedef a couple times, however it is already intrinsically contained in the `result_t` of the Reaction counts observable. I propose something as
```cpp
// in ReactionCounts.h
using counts_order1 = typename std::tuple_element<0, result_t>::type;
using counts_order2 = typename std::tuple_element<1, result_t>::type;
// in {SCPU/CPU/CPUD/....}StateModel / ....
using reaction_counts_order1_map = readdy::model::observables::ReactionCounts::counts_order1;
```